sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -597,7 +600,11 @@ public void setBeforeBreadcrumb(@Nullable BeforeBreadcrumbCallback beforeBreadcr
    * @return the cache dir. path or null if not set
    */
   public @Nullable String getCacheDirPath() {
-    return cacheDirPath;
+    if (cacheDirPath == null || cacheDirPath.isEmpty()) {
+      return null;
+    }
+
+    return dsnHash != null ? new File(cacheDirPath, dsnHash).getAbsolutePath() : cacheDirPath;
```

Comment:
in what case can we have dsnHash here? I understand it's nullable but is it a valid execution path to get this thing called before we get a hold of a dsn? Would it be fair to return `null` or log with the diagnostic logger in that case? Just a way for us to find out this is happening

Reply:
in case `dsn` is empty and SDK is disabled, `dsnHash` is null, SDK is NoOp anyway but we have to double-check due to NPE static code analysis.